Mathlib/NumberTheory/ModularForms/LevelOne.lean

```diff
@@ -0,0 +1,72 @@
+/-
+Copyright (c) 2024 Chris Birkbeck. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Birkbeck
+-/
+import Mathlib.NumberTheory.Modular
+import Mathlib.NumberTheory.ModularForms.SlashInvariantForms
+import Mathlib.NumberTheory.ModularForms.CongruenceSubgroups
+
+/-!
+# Level one modular forms
+
+This file contains results specific to modular forms of level one, ie. modular forms for `SL(2, ℤ)`.
+
+TODO: Add finite dimisionality of these spaces of modular forms.
+
+-/
+
+open UpperHalfPlane ModularGroup SlashInvariantForm ModularForm CongruenceSubgroup Complex
+local notation "SL(" n ", " R ")" => Matrix.SpecialLinearGroup (Fin n) R
+local notation  "Γ(" n ")"  => CongruenceSubgroup.Gamma n
+
+/--The map from `SL(2, ℤ)` to `Γ(1)`. Its preferable to work with this, since levels of modular
+forms are terms of type Subgroup `SL(2, ℤ)`. -/
+@[coe]def CongruenceSubgroup.coe1 : SL(2, ℤ) → Γ(1) :=
+  fun g => ⟨↑g, by simp [CongruenceSubgroup.Gamma_one_top]⟩
+
+instance : Coe SL(2, ℤ) Γ(1) := ⟨coe1⟩
```

Comment:
This worries me a bit since `Γ(1)` is by definition a subgroup of `SL(2, ℤ)`, so there will already be a `Coe` instance going the other way. Do we need it going this way as well? It sounds like it might cause confusion to have Coe instances in both directions at once.